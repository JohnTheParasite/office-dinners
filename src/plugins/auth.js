import AuthService from "@/services/authService";

export const Auth = {
  install(Vue) {
    Vue.prototype.$authService = AuthService;
  },
};