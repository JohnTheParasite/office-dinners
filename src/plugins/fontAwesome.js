import { library } from "@fortawesome/fontawesome-svg-core"
import {
  faArrowRightToBracket,
  faArrowUpRightFromSquare,
  faBan,
  faBurger,
  faChartLine,
  faCheck,
  faChevronDown,
  faChevronLeft,
  faChevronRight,
  faEye,
  faEyeSlash,
  faHandHoldingDollar,
  faHeart,
  faPencil,
  faPercent,
  faTrashCan,
  faUser,
  faUsers,
  faUtensils,
  faXmark
} from "@fortawesome/free-solid-svg-icons"
import { FontAwesomeIcon } from "@fortawesome/vue-fontawesome"
import Vue from "vue"

library.add(faChevronLeft)
library.add(faChevronRight)
library.add(faChevronDown)
library.add(faArrowUpRightFromSquare)
library.add(faCheck)
library.add(faPencil)
library.add(faTrashCan)
library.add(faBan)
library.add(faPercent)
library.add(faChartLine)
library.add(faUsers)
library.add(faBurger)
library.add(faHandHoldingDollar)
library.add(faUtensils)
library.add(faHeart)
library.add(faUser)
library.add(faArrowRightToBracket)
library.add(faXmark)
library.add(faEye)
library.add(faEyeSlash)
Vue.component("font-awesome-icon", FontAwesomeIcon)
