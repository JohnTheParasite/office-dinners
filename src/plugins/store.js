import store from "@/store/store"

export const Store = {
  install(Vue) {
    Vue.prototype.$store = store
  }
}
