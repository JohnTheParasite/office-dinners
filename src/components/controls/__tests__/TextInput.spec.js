import { createLocalVue, shallowMount } from "@vue/test-utils"
import textInput from "@/components/controls/TextInput"
import i18n from "@/i18n"
import IconNames from "@/enums/iconNames"
import "@/plugins/fontAwesome"

describe("TextInput", () => {
  const localVue = new createLocalVue()
  localVue.use(i18n)

  it("should render textInput properly", () => {
    let textInputElement = shallowMount(textInput, { localVue, i18n })
    expect(textInputElement).toMatchSnapshot()
  })

  it("should render with icon", () => {
    let textInputElement = shallowMount(textInput, { localVue, i18n, propsData: { icon: IconNames.EYE } })
    expect(textInputElement).toMatchSnapshot()
  })

  it("onIconClick property should work", () => {
    let onIconClickFn = jest.fn()
    let textInputElement = shallowMount(textInput, {
      localVue,
      i18n,
      propsData: { icon: IconNames.EYE, onIconClick: onIconClickFn }
    })
    textInputElement.vm.onIconClick()
    expect(onIconClickFn).toBeCalled()
  })

  it("getFaIcon should work", () => {
    let textInputElement = shallowMount(textInput, { localVue, i18n, propsData: { icon: IconNames.EYE } })
    expect(textInputElement.vm.getFaIcon).toBe("fa-solid fa-eye")
  })

  it("focusIn and focusOut should work", () => {
    let textInputElement = shallowMount(textInput, { localVue, i18n, propsData: { required: true, initValue: "" } })
    expect(textInputElement.vm.focused).toBeFalsy()
    textInputElement.vm.focusin()
    expect(textInputElement.vm.focused).toBeTruthy()
    textInputElement.vm.focusout()
    expect(textInputElement.vm.focused).toBeFalsy()
    expect(textInputElement.vm.errorMessage).not.toBe("")
  })

  it("click event should work", () => {
    let textInputElement = shallowMount(textInput, { localVue, i18n })
    let clickFunc = jest.fn()
    textInputElement.vm.$on("click", clickFunc)
    textInputElement.vm.click(1)

    expect(clickFunc).toBeCalledWith(1)
  })

  // it("input event should work", () => {
  //   let textInputElement = shallowMount(textInput, { localVue, i18n })
  //
  //   textInputElement.setProps({debounce: false})
  //   let inputFunc = jest.fn()
  //   textInputElement.vm.$on("input", inputFunc)
  //   textInputElement.vm.input(1)
  //
  //   expect(inputFunc).toBeCalledWith(1)
  // })

  it("getFaIcon should work", () => {
    let textInputElement = shallowMount(textInput, { localVue, i18n })
    textInputElement.setProps({ value: "blahBlah" })
  })
})
