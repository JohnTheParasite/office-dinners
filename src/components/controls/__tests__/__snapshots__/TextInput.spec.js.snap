// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput should render textInput properly 1`] = `
<div role="group" class="form-group text"><label for="2" class="input-label" style="display: none;"></label>
  <div role="group" class="input-group">
    <div class="form-group-input-container"><input id="2" name="" placeholder="" type="text" class="form-control">
      <!---->
    </div> <small class="text-error" style="display: none;"></small>
  </div>
</div>
`;

exports[`TextInput should render with icon 1`] = `
<div role="group" class="form-group text"><label for="4" class="input-label" style="display: none;"></label>
  <div role="group" class="input-group icon">
    <div class="form-group-input-container"><input id="4" name="" placeholder="" type="text" class="form-control"> <span class="input-icon"><font-awesome-icon-stub icon="fa-solid fa-eye"></font-awesome-icon-stub></span></div> <small class="text-error" style="display: none;"></small>
  </div>
</div>
`;
